### DISCUSSION TITLE
What is "Hero Rank" and "Role Rank" fields in filters?

### DISCUSSION THREAD

**Discussion by TheCrossCarrier** [Heroes-Profile/heroesprofile#964] [Q&A] [answered]
![image](https://github.com/Heroes-Profile/heroesprofile/assets/55375636/a6381e55-0ebe-4bad-9bc0-fd391625ab51)

**Comment by Zemill** [comment, marked as answer]
Ranks on heroes profile are based off our internal MMR algorithm and then bucketed into the different ranks based on percentiles.

So when looking at Player Rank filters, you are filtering on players MMR that matches the Heroes Profile Diamond rank for example

If you filter on Hero Rank, what we have done for every player is created a new MMR that only applies to that hero for that player in the particular mode.  This Hero Rank then goes up and down as expected with wins and losses, but it is compared against other players Hero Ranks.

So for an example, you might be a Gold level player based on your Player MMR, but you are really good at The Lost Vikings.  You have an 80% win rate with The Lost Vikings. So you might be a master level The Lost Vikings Player.

The same is done for roles.

Does that explain it?
